core/vibes/soul/sections/products-list-section/sorting.tsx

```diff
@@ -1,49 +1,77 @@
 'use client';
 
 import { parseAsString, useQueryState } from 'nuqs';
-import { Suspense, use } from 'react';
+import { Suspense, use, useOptimistic } from 'react';
 
 import { Select } from '@/vibes/soul/form/select';
+import { Streamable, useStreamable } from '@/vibes/soul/lib/streamable';
+
+import { ProductListTransitionContext } from './context';
 
 export type Option = {
   label: string;
   value: string;
 };
 
-type Props = {
-  options: Option[] | Promise<Option[]>;
+export function Sorting({
+  label = 'Sort',
+  options,
+  paramName = 'sort',
+  defaultValue = '',
+}: {
   label?: string;
+  options: Streamable<Option[]>;
   paramName?: string;
   defaultValue?: string;
-};
-
-export function Sorting({ label = 'Sort', options, paramName }: Props) {
+}) {
   return (
-    <Suspense fallback={<SortingSkeleton placeholder={label} />}>
-      <SortingInner label={label} options={options} paramName={paramName} />
+    <Suspense fallback={<SortingSkeleton />}>
+      <SortingInner
```

Comment:
Why aren't you using `mapStreamable` here?

Reply:
I didn’t refactor everything yet. Probably should at some point after 11/11